### PR TITLE
[stocks] adds stocks plugin

### PR DESCRIPTION
**Pull request by c650**

### CODE REVIEW

**Comment by c650** [comment]
requires an api key

File: src/app/plugins/stock.plugin.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    super();
+  }
+
+  public validate(message: IMessage, args: string[]) {
+    // in order to error handle
+    return args.length > 0;
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    if (!args || args.length < 1) {
```
Comment (joey-colon) [inline src/app/plugins/stock.plugin.ts]: Redundant since `validate()` check

File: src/app/plugins/stock.plugin.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      Environment.StockApiToken;
+
+    const data = await this.container.httpService
+      .get(call_url)
+      .then((res: IHttpResponse) => {
+        return res.data;
+      })
+      .catch((err) => {
+        console.log(err);
+        return new Object();
```
Comment (joey-colon) [inline src/app/plugins/stock.plugin.ts]: why are we returning a new object here?

File: src/app/plugins/stock.plugin.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      });
+
+    const quote = data['Global Quote'];
+
+    if (!quote) {
+      console.error('Query of ' + args.join(' ') + ' got ' + data);
```
Comment (joey-colon) [inline src/app/plugins/stock.plugin.ts]: nit: remove console calls

File: src/app/plugins/stock.plugin.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      message.reply('Query of `' + args.join(' ') + '` resulted in an error.');
+      return;
+    }
+
+    console.log(quote);
```
Comment (joey-colon) [inline src/app/plugins/stock.plugin.ts]: nit: remove

File: src/app/plugins/stock.plugin.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        'https://www.netclipart.com/pp/m/59-594517_arrow-going-up-png-stock-market-graph-up.png'
+      );
+    }
+
+    embed.setTitle(quote['01. symbol'] + ' @ ' + quote['05. price']);
+    embed.setFooter('data from https://www.alphavantage.co/');
+
+    embed.addField('Change', direction + this._formatNum(quote['10. change percent']) + '%', false);
+
+    if (args[1] == '-ohl') {
```
Comment (joey-colon) [inline src/app/plugins/stock.plugin.ts]: `===`

File: src/app/plugins/stock.plugin.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  private _API_URL: string = 'https://www.alphavantage.co/query?';
+  private _ADVANCED_QUOTE_LINK: string = 'https://finance.yahoo.com/quote/';
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public validate(message: IMessage, args: string[]) {
+    // in order to error handle
+    return args.length > 0;
```
Comment (joey-colon) [inline src/app/plugins/stock.plugin.ts]: fyi: you can do `!args.length` since 0 is a falsey value in js

File: src/app/plugins/stock.plugin.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+      '&apikey=' +
+      Environment.StockApiToken;
+
+    const data = await this.container.httpService
+      .get(call_url)
+      .then((res: IHttpResponse) => {
+        return res.data;
+      })
+      .catch((err) => {
+        console.log(err);
```
Comment (joey-colon) [inline src/app/plugins/stock.plugin.ts]: nit: remove console

**Comment by joey-colon** [comment]
also @c650 make sure you update the `.env.sample` file

File: src/app/plugins/stock.plugin.ts (hunk 6 above)
Comment (c650) [reply src/app/plugins/stock.plugin.ts]: less meaningful

File: src/app/plugins/stock.plugin.ts (hunk 2 above)
Comment (c650) [reply src/app/plugins/stock.plugin.ts]: the `then` returns an object, so i want to be consistent

**Pull request merged**